### PR TITLE
Codecov

### PR DESCRIPTION
**Pull request by silentroach**
What do you think about enabling codecov back?

### CODE REVIEW

File: test/github-issues/2875/issue-2875.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ describe("github issues > #2875 runMigrations() function is not returning a list
     let connections: Connection[];
     before(async () => connections = await createTestingConnections({
         entities: [__dirname + "/entity/*{.js,.ts}"],
-        migrations: [__dirname + "/migration/*.js"],
+        migrations: [__dirname + "/migration/*{.js,.ts}"],
```
[the remark this answers is not in the document]
Comment (silentroach) [reply test/github-issues/2875/issue-2875.ts]: no, migration is in .ts here

**Comment by silentroach** [comment]
>c8 uses bleeding edge Node.js features, make sure you're running Node.js >= 10.12.0.

seems like it is not possible now until there is compatibility with previous node versions

**Comment by pleerock** [comment]
Personally I find code coverage a funny toy. Sometimes it can be useful, in the cases when you want to improve test coverage and take a look what components aren't covered, but its not our case. Also its not possible to cover this codebase with 100% and even if it could be possible - its pointless. 

For those who needs test coverage - but its better idea to create a separate command that runs it? Or even `codecov` we have should be enough? 

But I'm not against merging this, lets ask what others think.

**Comment by silentroach** [comment]
code coverage just helps to write tests

**Comment by Kononnable** [comment]
You can check `typeorm-model-generator` - It's working with codecov and travis just fine, so it might help to figure out why does it hangs here.

As for test coverage. Some companies use it to decide which library to use. There is also one more good thing for us - it could be configured to fail when code coverage is below some point. It will automatically detect if someone forgot to remove `.only` from tests.

**Comment by pleerock** [comment]
> Some companies use it to decide which library to use

as for me its a stupid criteria to choose. Of course test coverage is important. But how do you determine if it's worth using? What if its let's say 50%? It's not worth using? 

> There is also one more good thing for us - it could be configured to fail when code coverage is below some point.

I had this experience and it was terrible. There are many cases when its not work properly. For example user improves code by removing some lines and tests for them and just adds his own functionality and test for it - result can be a fail because removed code and their test can affect overall coverage percent.

> It will automatically detect if someone forgot to remove .only from tests.

yes this one can be useful. Not sure however if any linter or extension to it can cover it.

**Comment by silentroach** [comment]
Why black (100%) and white (no coverage), why not something in between?

Coverage is a great tool to find some branches that are not covered by tests, this information is helpful very often. It is not about percentage as a number.

**Comment by silentroach** [comment]
I closed it just to investigate why is it hangs first, will reopen it later.

File: package.json [hunk 2]
```diff
@@ -63,9 +63,9 @@
     "mssql": "^4.2.2",
     "mysql": "^2.15.0",
     "mysql2": "^1.5.3",
+    "nyc": "^13.1.0",
     "pg": "^7.4.0",
     "redis": "^2.8.0",
-    "remap-istanbul": "^0.11.0",
```
[the remark this answers is not in the document]
Comment (silentroach) [reply package.json]: thanks, missed it :(

**Comment by rustamwin** [comment]
What about this way

```yaml
scripts:
- npm install codecov -g

after_success:
- codecov
```

**Comment by Kononnable** [comment]
Why was it closed?

**Comment by silentroach** [comment]
Lack of interest, it was finished more than a week ago.

I will try to make something more useful for all :)

**Comment by silentroach** [comment]
mm, can't reopen after force push

**Pull request closed without merging**